src/matrices.jl

```diff
@@ -351,23 +165,57 @@ function JointRecurrenceMatrix(x, y, ε; kwargs...)
     return JointRecurrenceMatrix(rm1.data .* rm2.data)
 end
 
+"""
+    JointRecurrenceMatrix(R1, R2; kwargs...)
+Create a joint recurrence matrix from given recurrence matrices `R1, R2`.
+"""
+function JointRecurrenceMatrix(R1::AbstractRecurrenceMatrix, R2::AbstractRecurrenceMatrix; kwargs...)
+    R3 = R1.data .* R2.data
+    return JointRecurrenceMatrix(R3)
+end
 
 ################################################################################
 # Scaling / fixed rate
 ################################################################################
 # here args... is (x, y, metric, ε) or just (x, metric, ε)
-function resolve_scale(args...; scale=1, fixedrate=false)
+function resolve_scale(args...; scale=1, fixedrate=false, fan=false)
     ε = args[end]
+    x = args[1]
+    if length(args) > 3
+        y = args[2]
+        metric = args[3]
+    else
+        metric = args[2]
+        y = x
+    end
     # Check fixed recurrence rate - ε must be within (0, 1)
-    if fixedrate
+    if fixedrate && (fan == false)
         sfun = (m) -> quantile(vec(m), ε)
         return resolve_scale(Base.front(args)..., 1.0; scale=sfun, fixedrate=false)
+    elseif (fixedrate && fan) || (fixedrate == false && fan)
+        return get_fan_threshold(x, y, metric, ε)
```

Comment:
I'm not happy with this: if the keyword argument `fan` is false, `resolve_scale` will return a number, but if it is `true`, it will return a vector, thus making the whole thing type-unstable.

Reply:
yepp, you are absolutely right. I followed your 1st suggestion.